### ISSUE TITLE
Invalid resolver signature/interface generated for map[string]interface{} list as field argument

### ISSUE THREAD

**Issue by orktes** [99designs/gqlgen#2041]
### What happened?

When creating a graphql schema such as

```graphql
directive @goModel(
  model: String
  models: [String!]
) on OBJECT | INPUT_OBJECT | SCALAR | ENUM | INTERFACE | UNION

input SomethingMappedToMap @goModel(model: "map[string]interface{}") {
  foo: String!
}

type Query {
  somethingList(arrayOfSomething: [SomethingMappedToMap!]!): String!
}
```

gqlgen generates incorrect (IMO) go interface for the QueryResolver

```go
type QueryResolver interface {
        SomethingList(ctx context.Context, arrayOfSomething map[string]interface{}) (string, error)
}
```

### What did you expect?

I was expecting an interface such as

```go
type QueryResolver interface {
        SomethingList(ctx context.Context, arrayOfSomething []map[string]interface{}) (string, error)
}
```

### Minimal graphql.schema and models to reproduce

```graphql
directive @goModel(
  model: String
  models: [String!]
) on OBJECT | INPUT_OBJECT | SCALAR | ENUM | INTERFACE | UNION

input SomethingMappedToMap @goModel(model: "map[string]interface{}") {
  foo: String!
}

type Query {
  somethingList(arrayOfSomething: [SomethingMappedToMap!]!): String!
}
```

### versions
 - gqlgen version: 0.17.1
 - go version: 1.16.7

**Comment by orktes** [comment]
Seems like this is a duplicate of https://github.com/99designs/gqlgen/issues/1406. Seems like fairly critical bug so will leave open for the deserved visibility ;)

**Comment by frederikhors** [comment]
@orktes I need to close this because we have many issues open.

**Comment by sarmad4444** [comment]
Blocked by this issue too. Any update?

**Comment by frederikhors** [comment]
> Blocked by this issue too. Any update?

Post a repro project, please.
